spring-integration-core/src/main/java/org/springframework/integration/dsl/RecipientListRouterSpec.java

```diff
@@ -89,8 +97,14 @@ public RecipientListRouterSpec recipientMessageSelector(String channelName, Mess
 	 * @return the router spec.
 	 */
 	public <P> RecipientListRouterSpec recipient(String channelName, GenericSelector<P> selector) {
-		Assert.hasText(channelName);
-		((DslRecipientListRouter) this.target).add(channelName, selector);
+		MessageSelector messageSelector;
+		if (selector instanceof MessageSelector) {
+			messageSelector = (MessageSelector) selector;
+		}
+		else {
+			messageSelector = new MethodInvokingSelector(new LambdaMessageProcessor(selector, null));
```

Comment:
I realize the `LMP` will work for any functional interface. My point is, this won't work...

	.recipient("foo", new Foo())

	public static class Foo implements GenericSelector<Message<?>> {

		@Override
		public boolean accept(Message<?> source) {
			return false;
		}

		public boolean foo(String bar) {
			return false;
		}

	}

...because the LMP will select the second method not the first.